### PR TITLE
Wrap "Pattern too complex" exception into an IllegalArgumentException

### PR DESCRIPTION
**Pull request by afoucret**
The behavior of `java.util.regex.Pattern.:compile` has been updated in a way that make it possible too handle OutOfMemoryError errors cause by too complex pattern more gracefully and make them recoverable.

### CODE REVIEW

File: modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml [hunk 1]
```diff
@@ -44,6 +44,32 @@
...
+"Too complex regex pattern":
+    - do:
+        catch: bad_request
+        indices.create:
+          index: test_too_complex_regex_pattern
+          body:
+            settings:
+              index:
+                analysis:
+                  analyzer:
```
Comment (saikatsarkar056) [inline modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml:57]: What is the rectangle shaped char at the end of each line? Can we remove them?

File: docs/changelog/109173.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pr: 109173
+summary: Wrap "Pattern too complex" exception into an `IllegalArgumentException`
```
Comment (kderusso) [inline docs/changelog/109173.yaml:2]: Did this go live in an Elasticsearch release? If not I don't think it needs a changelog.

**Review by kderusso — APPROVED** [review]
Looks reasonable to me with one question about release notes.

File: modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml (hunk 1 above)
Comment (ioanatia) [reply modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml:57]: GH does not render it well - but this is an actual case where we know `Pattern::compile` throws an `OutOfMemoryError` so I think it is fine to keep it. Might be worth adding a comment here for folks that might read this code in the future? @afoucret

**Review by ioanatia — APPROVED** [review]

File: modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml (hunk 1 above)
Comment (afoucret) [reply modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/indices.analyze/10_analyze.yml]: I moved this test to a more appropriate place and added a comment to it

**Pull request merged**